google-cloud-bigtable/lib/google/cloud/bigtable/instance_admin_client.rb

```diff
@@ -0,0 +1,424 @@
+# frozen_string_literal: true
+
+require "google/cloud/bigtable/admin/v2"
+require "google/bigtable/admin/v2/instance_pb"
+
+module Google
+  module Cloud
+    module Bigtable
+      # Protobuf aliases
+      Instance = Google::Bigtable::Admin::V2::Instance
+      Cluster = Google::Bigtable::Admin::V2::Cluster
+
+      class InstanceAdminClient
+        # Bigtable instance admin client for creating, configuring, and deleting
+        # instances and clusters
+
+        attr_reader :project_id, :options
+
+        # @param project_id [String]
+        # @param options [Hash]
+        def initialize project_id, options = {}
+          @project_id = project_id
+          @options = options
+        end
+
+        # Get instance information
+        #
+        # @param instance_id [String]
+        # @return [Google::Bigtable::Admin::V2::Instance]
+        # @example
+        #   require "google/cloud/bigtable"
+        #
+        #   client = Google::Cloud.bigtable(
+        #     project_id: "project-id",
+        #     client_type: :instance
+        #   )
+        #
+        #   client.instance("my-instance-id")
+
+        def instance instance_id
+          client.get_instance(
+            instance_path(instance_id)
+          )
+        end
+
+        # List all instances
+        #
+        # @param page_token [String]
+        #   The value of +next_page_token+ returned by a previous call.
+        # @return [Google::Bigtable::Admin::V2::ListInstancesResponse]
+        #   Response has list of instances, failed locations and next page token
+        # @example
+        #   require "google/cloud/bigtable"
+        #
+        #   client = Google::Cloud.bigtable(
+        #     project_id: "project-id",
+        #     client_type: :instance
+        #   )
+        #
+        #   response = client.instances
+        #   instances = response.instances
+        #
+        #   if response.failed_locations.any?
+        #     puts response.failed_locations
+        #   end
+        #
+        #   # Get instances using next page token
+        #   unless response.next_page_token.empty?
+        #     response = client.instances(response.next_page_token)
+        #     instances.concat(response.instances)
+        #   end
+
+        def instances page_token: nil
+          client.list_instances(
+            project_path,
+            page_token: page_token
+          )
+        end
+
+        # Create instance with clusters.
+        #
+        # @param instance [Bigtable::Instance | Hash]
+        #   The instance to create. Fields are name, type and lables.
+        #   Serve nodes cannot be specified for development instances
+        # @param clusters [Array<Bigtable::Cluster>]
```

Comment:
`Bigtable::Cluster` doesn't exist. Should be `Google::Bigtable::Admin::V2::Cluster` instead.